docs/general/dql/selects.rst

```diff
@@ -740,6 +740,45 @@ an upper-bound::
     Both the ``from`` and ``to`` index values are inclusive.
     Using an index greater than the array size results in an empty array.
 
+For selecting records having values with empty or unassigned arrays, like::
+
+    cr> insert into locations (id, name, position, kind, landmarks) values
+    ... (99, 'Voidspace', 0, 'Milliways', []);
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh locations
+
+    cr> refresh table locations;
+    REFRESH OK, 1 row affected (... sec)
+
+a suitable expression to match for both ``NULL`` values and empty arrays, is::
+
+    cr> select count(*) from locations where landmarks is null;
+    +----------+
+    | count(*) |
+    +----------+
+    |       14 |
+    +----------+
+    SELECT 1 row in set (... sec)
+
+.. Hidden: drop record again
+
+    cr> delete from locations where id=99;
+    DELETE OK, 1 row affected  (... sec)
+
+For determining the number of items in an array, you can use the
+:ref:`array_length <scalar-array_length>` scalar function::
+
+    cr> select count(*) from locations where
+    ... array_length(landmarks, 1) = 2;
+    +----------+
+    | count(*) |
+    +----------+
+    |        1 |
+    +----------+
+    SELECT 1 row in set (... sec)
+
```

Comment:
```suggestion
.. NOTE::
    
    :ref:`array_length <scalar-array_length>` will return ``NULL`` on empty arrays.
```

Reply:
Very good suggestion to improve understanding in this area.